opensearch/src/main/java/org/opensearch/sql/opensearch/mapping/IndexMapping.java

```diff
@@ -50,55 +59,42 @@ public int size() {
    * @return field type in string. Or null if not exist.
    */
   public String getFieldType(String fieldName) {
-    return fieldMappings.get(fieldName);
-  }
-
-  /**
-   * Get all field types and transform raw string type to expected type.
-   *
-   * @param transform transform function to transform field type in string to another type
-   * @param <T> expected field type class
-   * @return mapping from field name to field type
-   */
-  public <T> Map<String, T> getAllFieldTypes(Function<String, T> transform) {
-    return fieldMappings.entrySet().stream()
-        .collect(Collectors.toMap(Map.Entry::getKey, e -> transform.apply(e.getValue())));
+    if (!fieldMappings.containsKey(fieldName)) {
+      return null;
+    }
+    return fieldMappings.get(fieldName).typeName();
   }
 
   @SuppressWarnings("unchecked")
-  private Map<String, String> flatMappings(Map<String, Object> indexMapping) {
-    ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
-
-    flatMappings(
-        ((Map<String, Object>) indexMapping.getOrDefault("properties", emptyMap())),
-        "",
-        builder::put);
-    return builder.build();
+  private Map<String, OpenSearchDataType> parseMapping(Map<String, Object> indexMapping) {
```

Comment:
In general,  I think we could define OpenSearchMapping class which provide resolve method.
```
interface OpenSearchMapping {
   OpenSearchDataType resolve(Identifier id);
}
```
The benfit is resolve interface provide more flexibility, for example,
1.  `manager`, should be resolved as object.
2. `manager`.'age', should be resolve as integer.
3. `manager`.`name`, should be resolve as object.
```
PUT my-index-000001/_doc/1
{ 
  "region": "US",
  "manager": { 
    "age":     30,
    "name": { 
      "first": "John",
      "last":  "Smith"
    }
  }
}
```

Reply:
Added in 84cbc673.